### ISSUE TITLE
False positive: `matplotlib.figure.Figure.subplots()` identified as returning `ndarray` instead of `Axes`

### ISSUE THREAD

**Issue by wshanks** [PyCQA/pylint#8261]
### Bug description

In an environment with `matplotlib`, the following code:

```
"""Example"""
from matplotlib.figure import Figure


def func():
    """Test"""
    figure = Figure()
    axes = figure.subplots()
    return axes.get_figure()
```

produces

```
/tmp.py:9:11: E1101: Instance of 'ndarray' has no 'get_figure' member (no-member)
```

### Command used

```shell
pylint tmp.py
```

### Pylint output

```shell
/tmp.py:9:11: E1101: Instance of 'ndarray' has no 'get_figure' member (no-member)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.10.6
```

### OS / Environment

Fedora 36

### Additional dependencies

matplotlib==3.5.3
numpy==1.23.3

**Comment by wshanks** [comment]
Is there a way to work around this other than either disabling `no-member` entirely or disabling it for every part of the code that uses the return value of `func()`?

**Comment by wshanks** [comment]
I am closing this because it reminds me of https://github.com/PyCQA/pylint/issues/3601 -- the code is too complex for pylint to follow -- and the workaround is to use `squeeze=False` so that the return type is consistent.

**Comment by pamelafox** [comment]
I just ran into the same issue with this code:

```
    axes:Axes = fig.subplots()
    print(type(axes)). # <class 'matplotlib.axes._axes.Axes'>
    axes.plot([1, 2]). # Instance of 'ndarray' has no 'plot' memberpylint(no-member)
```

The workaround you suggest does work:

```
    axes:Axes = fig.subplots(squeeze=False)[0][0]
```

But its non-ideal. Anyway, just putting the code here to help others.

**Comment by wshanks** [comment]
Perhaps using `fig.add_subplot()` is a cleaner solution than `fig.subplots(squeeze=False)[0][0]` for indicating that only a single subplot is wanted.

**Comment by pamelafox** [comment]
Ah, thank you for that suggestion! That seems to work and is cleaner. I'm new to matplotlib and didn't realize that was an option as well.
